src/Features/CSharp/Portable/EditAndContinue/SyntaxUtilities.cs

```diff
@@ -83,7 +83,7 @@ public static void AssertIsBody(SyntaxNode syntax, bool allowLambda)
             if (LambdaUtilities.IsLambdaBody(syntax))
             {
                 Debug.Assert(allowLambda);
-                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax);
+                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax || syntax is ArrowExpressionClauseSyntax);
```

Comment:
> || syntax is ArrowExpressionClauseSyntax [](start = 80, length = 41)

I would rather change IsLambdaBody instead. It'd be more consistent with lambdas if we considered the actual expression the body of a local function rather then the arrow expression.